File: Data-Structure/JS/sort-insertion-sort.js
const arr = [0, 1, 5, 4, 3, 2, 6];
const insertionSort = function (arr) {
    if (arr.length < 2) return arr;

    for (let i = 0; i < arr.length; i++) {
        let smallIdx = i;
        for (let j = i; j < arr.length; j++) {
            if (arr[smallIdx] > arr[j]) {
                smallIdx = j;
            }
        }
        [arr[i], arr[smallIdx]] = [arr[smallIdx], arr[i]];
    }
    return arr;
}

console.log(insertionSort(arr));

File: Data-Structure/Python/sort/sort-insertion-sort.py
def insertionSort(arr):
    for i in range(1, len(arr)):
        current = arr[i]
        for j in range(i-1, -1, -1):
            print(f"current = {current}")
            print(f"current j = {arr[j]}")
            if (current < arr[j]):
                arr[j+1] = arr[j]
            else:
                break

        arr[j + 1] = current


# arr = [12, 11, 13, 5, 6]
arr = [11, 13, 5]
insertionSort(arr)
print("Sorted array is:")
for i in range(len(arr)):
    print("%d" % arr[i])
